### PR TITLE
New API Endpoint: users/list.

### PR DESCRIPTION
**Pull request by Lucas-Ellenberger**
This is the python interface CLI / API usage and tests for the new users/list ~~and courses/users/list APIs~~. (Edit: Outdated)

### CODE REVIEW

File: autograder/cli/common.py [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
+USER_HEADERS = ['email', 'name']
+
+COURSE_USER_HEADERS = USER_HEADERS + ['role', 'lms-id']
+SYNC_HEADERS = COURSE_USER_HEADERS + ['operation']
+
+BASE_SERVER_USER_HEADERS = USER_HEADERS + ['role']
+SERVER_USER_HEADERS = BASE_SERVER_USER_HEADERS + ['courses']
+COURSE_INFO_HEADERS = ['id', 'name', 'role']
+
+INDENT = '    '
```
Comment (BatuhanSA) [inline autograder/cli/common.py:10]: Can you lexicographically re-order the constants, as that format is consistent throughout the repo

Comment (Lucas-Ellenberger) [reply autograder/cli/common.py:10]: Since the constants build off one another, lexicographically re-ordering them would not work.

**Comment by BatuhanSA** [comment]
lgtm!

**Review by BatuhanSA — APPROVED** [review]
lgtm!

**Review by eriq-augustine — CHANGES_REQUESTED** [review]
Looking pretty good.
There's some work to be done, but the core is good.

@BatuhanSA, also make sure to read these comments closely.

File: autograder/cli/common.py [hunk 2]
```diff
@@ -8,33 +17,100 @@
...
+def _list_server_users_table_normalize(users, header = True, keys = BASE_USER_HEADERS):
+    rows = []
     for user in users:
+        if (user['type'] != SERVER_USER_INFO_TYPE):
+            raise ValueError("Invalid user type for listing normalized server users table:"
+                    + " Expected: '%s', actual: '%s'.", SERVER_USER_INFO_TYPE, user['type'])
+
         row = [user[key] for key in keys]
-        print("\t".join([str(value) for value in row]))
+        if ((user.get('courses') is None) or (len(user['courses']) == 0)):
```
Comment (eriq-augustine) [inline autograder/cli/common.py:104]: This comment requires no action.

For cases like this, I often prefer:
```
    if (len(user.get('courses', [])) == 0):
```

Note that your version also protects against null/nil/None values.

**Review by eriq-augustine — CHANGES_REQUESTED** [review]
Nice, just about there.
Just a small spacing issue and some documentation work.

**Review by eriq-augustine — APPROVED** [review]
lgtm

**Pull request merged**